packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -312,66 +497,88 @@ export default {
-          eventBranches.some((branch) => e.id >= branch[0] && e.id <= branch[1])
+      if (this.filters.declutter.hideUnlabeled.on) {
+        events = events.filter(
+          (e) =>
+            e.labels.size > 0 || e.codeObject.type !== CodeObjectType.FUNCTION
         );
-
-        // it's necessary to remove package wrap in Component Diagram,
-        // without this code package will be rendered in collapsed state
-        if (
-          this.rootCodeObject.type === 'class' &&
-          this.rootCodeObject.packageObject
-        ) {
-          classMap = new ClassMap([
-            classMap.codeObjects.filter(
-              (co) => co.id !== this.rootCodeObject.packageObject.id
-            ),
-          ]);
-        }
       }
 
-      events = events.filter(
-        (e) =>
-          this.filters.unlabeledEvents.on ||
-          e.labels.size > 0 ||
-          e.codeObject.type !== CodeObjectType.FUNCTION
-      );
+      if (
+        this.filters.declutter.hideElapsedTimeUnder.on &&
+        this.filters.declutter.hideElapsedTimeUnder.time > 0
+      ) {
+        events = events.filter(
+          (e) =>
+            e.returnEvent &&
+            e.returnEvent.elapsedTime &&
+            e.returnEvent.elapsedTime >=
+              this.filters.declutter.hideElapsedTimeUnder.time / 1000
+        );
+      }
 
-      if (this.filters.hideMediaRequests.on) {
-        const mediaRegex = [
-          'application/javascript',
-          'application/ecmascript',
-          'audio/.+',
-          'font/.+',
-          'image/.+',
-          'text/javascript',
-          'text/ecmascript',
-          'text/css',
-          'video/.+',
-        ].map((t) => new RegExp(t, 'i'));
-        const excludedEvents = [];
-        events.forEach((e) => {
-          if (
-            e.http_server_response &&
-            e.http_server_response.mime_type &&
-            mediaRegex.some((regex) =>
-              regex.test(e.http_server_response.mime_type)
-            )
-          ) {
-            excludedEvents.push(e.parent_id);
+      if (
+        this.filters.declutter.hideName.on &&
+        this.filters.declutter.hideName.names.length
+      ) {
+        this.filters.declutter.hideName.names.forEach((fqid) => {
+          const codeObject = classMap.codeObjects.filter(
+            (co) => co.fqid === fqid
+          )[0];
+          if (codeObject) {
+            events = events.filter((e) => !codeObject.allEvents.includes(e));
           }
         });
-        events = events.filter((e) => !excludedEvents.includes(e.id));
+      }
+
+      let eventIds = [];
+      if (this.traceFilterValue) {
+        const filterValue = this.traceFilterValue.trim().split(' ');
+        if (filterValue.length) {
+          filterValue.forEach((item) => {
+            if (item.startsWith('event:')) {
+              eventIds.push(parseInt(item.replace('event:', ''), 10));
+            }
+          });
+        }
+      } else {
```

Comment:
This should be highlighting events in the trace, not filtering out non-matching events. See the animation here: https://github.com/applandinc/appmap-js/issues/419